File: Stacksusing two Queues/stackUsingTwoQueues.java

import java.io.*;
import java.lang.*;
import java.util.*;

class Main{
	//This class implements the stack data structure using two queues.
	static class Stack {
		static Queue<Integer> q1 = new LinkedList<Integer>();
		static Queue<Integer> q2 = new LinkedList<Integer>();
		static int curr_size;

		Stack(){
			curr_size = 0;
		}

		static void push(int x){
			curr_size++;
			q2.add(x);
			while (!q1.isEmpty()) {
				q2.add(q1.peek());
				q1.remove();
			}
			Queue<Integer> q = q1;
			q1 = q2;
			q2 = q;
		}
		static int pop(){
			if (q1.isEmpty())
				return -1;
            int ans =q1.peek();
			q1.remove();
			curr_size--;
            return ans;
		}
		static int top(){
			if (q1.isEmpty()) return -1;
			return q1.peek();
		}

		static int size(){
			return curr_size;
		}
	}

	//Main Function
    public static void main(String[] args){
		Stack stack = new Stack();
		stack.push(2);
		stack.push(3);
        System.out.println(stack.pop());
        stack.push(4);
        System.out.println(stack.pop());
        System.out.println(stack.top());
        System.out.println(stack.size());
	}
}

File: 15_Stacksusing two Queues/Ideal Solutions/stackUsingTwoQueues.js
class Stack {
  #Q1 = [];
  #Q2 = [];

  push(value) {
    if (this.#Q1.length !== 0) {
      while (this.#Q1.length) {
        this.#Q2.push(this.#Q1.shift());
      }
      this.#Q1.push(value);
      while (this.#Q2.length) {
        this.#Q1.push(this.#Q2.shift());
      }
    } else {
      this.#Q1.push(value);
    }
  }
  pop() {
    return this.#Q1.length ? this.#Q1.shift() : "empty";
  }
}

const stack = new Stack();
/* stack.push(9);
console.log(stack.pop());
console.log(stack.pop());
stack.push(10);
console.log(stack.pop());
 */